### PR TITLE
test: added serialization and equality tests for `NodeContent`

### PR DESCRIPTION
**Pull request by timcreatedit**
## Checklist

- [X] My PR title is in the style of [conventional commits](https://www.conventionalcommits.org/)
- [X] All public facing APIs are documented with [dartdoc](https://dart.dev/guides/language/effective-dart/documentation)
- [X] I have added tests to cover my changes

### CODE REVIEW

File: packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+          components: <String, Component>{},
+          componentSets: <String, ComponentSet>{},
+          schemaVersion: 1,
+          styles: <String, Style>{},
+        );
+
+        expect(nodeContent1, nodeContent2);
+      });
+
+      test('should not be equal if any property is different', () {
```
Comment (JesperBllnbm) [inline packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart:116]: ahm, is this a oversight? Seems to be the same test as above..

File: packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  group('NodeContent', () {
+    group('fromJson', () {
+      test('should return a valid model', () {
+        final Map<String, dynamic> jsonMap = {
+          'name': 'name',
+          'role': 'role',
+          'lastModified': '2021-08-06T00:00:00Z',
+          'thumbnailUrl': 'thumbnailUrl',
+          'version': 'version',
+          'document': null,
```
Comment (JesperBllnbm) [inline packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart:17]: Has document its own toJson and fromJson test?

File: packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart [hunk 3]
```diff
@@ -0,0 +1,147 @@
+import 'package:figma/figma.dart';
+import 'package:figma_variables_api/figma_variables_api.dart';
+import 'package:test/test.dart';
+
+void main() {
+  setUp(() {});
```
Comment (JesperBllnbm) [inline packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart:6]: Has this any effect?

**Review by JesperBllnbm — CHANGES_REQUESTED** [review]

File: packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart (hunk 3 above)
Comment (timcreatedit) [reply packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart:6]: Nope, should've been deleted!

File: packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart (hunk 1 above)
Comment (timcreatedit) [reply packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart:116]: Haha you're right, I just trusted copilot, mb

File: packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart (hunk 2 above)
Comment (timcreatedit) [reply packages/figma_variables_api/test/src/dto/nodes/node_content_test.dart:17]: Not really, but it's part of package:figma, so not our job here

**Pull request merged**